### PR TITLE
Dca 702 custom dataset filtering

### PR DESCRIPTION
**Pull request by SergeyMoroz0703**

### CODE REVIEW

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 1]
```diff
@@ -4,63 +4,391 @@ platform: platform
...
+{{% /warning %}}
+
+---
+## <a id="mainenvironmententerprise"></a> Enterprise-scale environment
+
+After adding your custom app-specific actions, you should now be ready to run the required tests for the Marketplace Data Center Apps Approval process. To do this, you'll need an **enterprise-scale environment**.
+
+### <a id="instancesetup"></a>5. Set up an enterprise-scale environment Jira Data Center on AWS
+
+We recommend that you use the [AWS Quick Start for Jira Data Center](https://aws.amazon.com/quickstart/architecture/jira/) to deploy a Jira Data Center enterprise-scale environment. This Quick Start will allow you to deploy Jira Data Center with a new [Atlassian Standard Infrastructure](https://aws.amazon.com/quickstart/architecture/atlassian-standard-infrastructure/) (ASI) or into an existing one.
```
Comment (mmizin) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:350]: How about "We recommend you to set up this development using the"

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 2]
```diff
@@ -4,63 +4,391 @@ platform: platform
...
 
 To reduce costs, we recommend you to keep your deployment up and running only during the performance runs.
 
-### AWS cost estimation ###
+#### AWS cost estimation
 [AWS Pricing Calculator](https://calculator.aws/) provides an estimate of usage charges for AWS services based on certain information you provide.
-Monthly charges will be based on your actual usage of AWS services, and may vary from the estimates the Calculator has provided.
+Monthly charges will be based on your actual usage of AWS services and may vary from the estimates the Calculator has provided.
+
+*The prices below are approximate and may vary depending on such factors like region, instance type, deployment type of DB, and other.
```
Comment (mmizin) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:370]: How about "ec2 instance type, db instance type"

File: docs/dc-apps-performance-toolkit-user-guide-confluence.md [hunk 3]
```diff
@@ -4,24 +4,342 @@ platform: platform
...
+
+{{% note %}}
+You are responsible for the cost of AWS services used while running this Quick Start reference deployment. This Quick Start doesn't have any additional prices. See [Amazon EC2 pricing](https://aws.amazon.com/ec2/pricing/) for more detail.
+{{% /note %}}
+
+To reduce costs, we recommend you to keep your deployment up and running only during the performance runs.
+
+#### AWS cost estimation for the development environment
+
+AWS Confluence Data Center development environment infrastructure costs about 10 - 15$ per working week depending on such factors like region, instance type, deployment type of DB, and other.
```
Comment (mmizin) [inline docs/dc-apps-performance-toolkit-user-guide-confluence.md:54]: How about "ec2 instance type, db instance type"

File: docs/dc-apps-performance-toolkit-user-guide-jira.md [hunk 4]
```diff
@@ -4,63 +4,391 @@ platform: platform
...
+
+{{% note %}}
+You are responsible for the cost of AWS services used while running this Quick Start reference deployment. This Quick Start doesn't have any additional prices. See [Amazon EC2 pricing](https://aws.amazon.com/ec2/pricing/) for more detail.
+{{% /note %}}
+
+To reduce costs, we recommend you to keep your deployment up and running only during the performance runs.
+
+#### AWS cost estimation for the development environment
+
+AWS Jira Data Center development environment infrastructure costs about 10 - 15$ per working week depending on such factors like region, instance type, deployment type of DB, and other.
```
Comment (mmizin) [inline docs/dc-apps-performance-toolkit-user-guide-jira.md:54]: How about "ec2 instance type, db instance type"

File: app/util/data_preparation/confluence_prepare_data.py [hunk 5]
```diff
@@ -75,6 +79,16 @@ def __get_pages(confluence_api, count):
     return pages
 
 
+def __get_custom_pages(confluence_api, count, cql):
+    pages = []
+    if cql:
+        pages = confluence_api.get_content_search(
+            0, count, cql=cql)
+        if not pages:
+            raise SystemExit(f"There are no Pages in Confluence could be find with CQL: {cql}")
```
[the remark this answers is not in the document]
Comment (ometelytsia) [reply app/util/data_preparation/confluence_prepare_data.py:88]: f"ERROR: There are no pages in Confluence could be found with CQL: {cql}"

File: docs/dc-apps-performance-toolkit-user-guide-bitbucket.md [hunk 6]
```diff
@@ -489,40 +491,67 @@ You can extend Selenium scripts to measure the end-to-end browser timings.
 
 We use **Pytest** to drive Selenium tests. The `bitbucket-ui.py` executor script is located in the `app/selenium_ui/` folder. This file contains all browser actions, defined by the `test_ functions`. These actions are executed one by one during the testing.
 
+#### Example of app-specific Selenium action development
+You develop an app that adds additional UI elements to repository page, in this case you should edit `dc-app-performance-toolkit/extension/bitbucket/extension_ui.py`:
```
[the remark this answers is not in the document]
Comment (SergeyMoroz0703) [reply docs/dc-apps-performance-toolkit-user-guide-bitbucket.md:495]: From my point of view here we'are talking about repositories generally but not about the particular repository. (im not sure)

[the remark this answers is not in the document]
Comment (ometelytsia) [reply docs/dc-apps-performance-toolkit-user-guide-bitbucket.md:495]: Any repository page, not a specific one. So `to a repository page`

File: docs/dc-apps-performance-toolkit-user-guide-jira.md (hunk 2 above)
Comment (SergeyMoroz0703) [reply docs/dc-apps-performance-toolkit-user-guide-jira.md:370]: This is a good idea, however, this syntax construction adds some boundaries that we can overstep somehow.

File: docs/dc-apps-performance-toolkit-user-guide-bitbucket.md [hunk 7]
```diff
@@ -21,7 +21,9 @@ To use the Data Center App Performance Toolkit, you'll need to:
 For simple spikes or tests, you can skip steps 1-2 and target any Bitbucket test instance. When you [set up your execution environment](#executionhost), you may need to edit the scripts according to your test instance's data set.
 {{% /note %}}
 
-## <a id="instancesetup"></a> Setting up Bitbucket Data Center
+---
```
Comment (ometelytsia) [inline docs/dc-apps-performance-toolkit-user-guide-bitbucket.md:24]: Need to bump the date on the 7th line of this file.

**Review by ometelytsia — APPROVED** [review]

**Review by mmizin — APPROVED** [review]

**Pull request merged**